ush/ioda/bufr2ioda/marine/b2i/b2iconverter/ocean.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import numpy as np
+import numpy.ma as ma
+import math
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import netCDF4 as nc
+import xarray as xr
+
+# OceanBasin class provides a facility to add an OceanBasin
+# metadata variable using lon and lat
+# basic definition of ocean basins is read from an nc file,
+# We search for the filename, depending on the system
+# The path to the ocean basin nc file can be supplied
+# in the implementation of the converter
+
+# the main method is GetStationBasin which returns the ocean basin
+# for a list of station coordinates
+# there are methods for plotting and printing the ocean basin data
+# as well as printing and plotting station basin data
+
+
+class OceanBasin:
+    def __init__(self):
+        self.FindOceanBasinNCFile()
+
+    def SetOceanBasinNCFilePath(self, filename):
+        self.ocean_basin_nc_file_path = filename
+
+    def FindOceanBasinNCFile(self):
+        # try on hera:
+        trypath = "/scratch2/NCEPDEV/ocean/Guillaume.Vernieres/data/static/common/RECCAP2_region_masks_all_v20221025.nc"
```

Comment:
this needs to be passed as an argument similarly to what is done for the netcdf converters:
See here for example, https://github.com/NOAA-EMC/GDASApp/blob/b853fe9b7f142df4cec2a191601be9f82be1c6fe/parm/soca/obsprep/obsprep_config.yaml#L9